testing/python/iree/testing.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 The IREE Authors
```

Comment:
Here for threading. Yeah we'll want to store large artifacts in our GCS bucket. We already have [some integration tests](https://github.com/iree-org/iree/blob/62627e80c0/integrations/tensorflow/test/python/iree_tfl_tests/mobilenet_v1_test.py) that fetch from there (I can upload to the bucket, once we figure out what we're doing here). This is also not ideal because it makes every test do its own network access and then flakes abound and when testing locally it unnecessarily fetches each time. At least if we use that approach though, we can stick to *one* bad way of doing things :stuck_out_tongue: It would be really nice if GitLFS was significantly more functional...

There's another problem here as well: mlir is not a stable serialization format. Even the bytecode format isn't stable when dialects change (the fortunately no longer breaks the world when someone changes the pretty printer). We really don't want opaque unstable formats for tests. To do that, we'd need an MLIR bytecode file containing only versioned StableHLO ops or similar. Is there a stable format we can use for this?

Reply:
The file is generate from a Jax model using IREE Jax. Doesn't this produce StableHLO? I don't know what its exact version.  I am outputting in the MLIR text format, because recently the bytecode format got a breaking change. I don't know the details there. My thinking was that if something changed recently it is likely to change in the future. I will change it to generate bytecode.

Reply:
I change it to bytecode.

Reply:
Can someone who has access upload the artifacts? I will change the test to download them.

Reply:
Uploaded to https://storage.googleapis.com/iree-model-artifacts/mnist_train.d13d480df48598695193bd1a20795a0deb57cc659004914a4bd484eab85d524f.tar

Reply:
I change the test to download the data on-the-fly.

Reply:
> The file is generate from a Jax model using IREE Jax. Doesn't this produce StableHLO?

I'm not sure about the IREE JAX pipeline, but JAX currently produces StableHLO by default. The APIs for creating stable artifacts are nearing completion (PRs in review), then we will be offering 1month forward/backward compat guarantees in H1 2023, and in H2 will flip to longer 5yr compatibility guarantees. That is to say, serialized StableHLO can be used in the longer term, but in the shorter term one of the solutions proposed by @GMNGeoffrey is probably best.